src/lib.rs

```diff
@@ -1065,29 +945,33 @@ mod test {
                                             \"size\": 100,
                                             \"mimeType\": \"text/html; charset=utf8\"
                                         },
-                                        \"redirectURL\": \"\",
-                                        \"headersSize\": -1,
-                                        \"bodySize\": -1
+                                        \"redirectUrl\": \"\",
+                                        \"headersSize\": null,
+                                        \"bodySize\": null
+                                    },
+                                    \"cache\": {
+                                        \"beforeRequest\": null,
+                                        \"afterRequest\": null
                                     },
-                                    \"cache\": {},
                                     \"time\": 15,
                                     \"timings\": {
-                                         \"blocked\": -1,
-                                         \"dns\": -1,
-                                         \"connect\": -1,
+                                         \"blocked\": null,
+                                         \"dns\": null,
+                                         \"connect\": null,
                                          \"send\": 4,
                                          \"wait\": 5,
                                          \"receive\": 6,
-                                         \"ssl\": -1
+                                         \"ssl\": null
                                     }
                                 }
                             ],
                             \"comment\": \"Comment\"
                         }";
-        assert_eq!(json::from_str(log_json),
-                   json::from_str(json::encode(&log).as_slice()));
+		let log_from_str: Log = serde_json::from_str(log_json).unwrap();
```

Comment:
nit: the indentation here is weird; I would expect it to match the assert_eq on the next line.